src/lib/lib.c

```diff
@@ -60,7 +61,12 @@ void *memset(void *s, int c, size_t n)
 int memcpy_s(void *dest, size_t dest_size,
 	     const void *src, size_t src_size)
 {
-	return arch_memcpy_s(dest, dest_size, src, src_size);
+	int ret = arch_memcpy_s(dest, dest_size, src, src_size);
+
+	/* Unconditionally crash in case arch_memcpy_s fails */
+	if (!ret)
+		panic(SOF_IPC_PANIC_ASSERT);
+	return ret;
```

Comment:
@paulstelian97 Isn't this overlapping with the other patch? I mean either you have a custom assert after each memcpy_s() (which is never left out of builds) or you have similar code here.

I'll let the SOF maintainers weigh in on the correct behaviour with mempcy errors in product builds.

Reply:
I have dropped my other suggestion actually, because there are two instances of memcpy_s (one for actual compiled code and one as a mock in cmocka). For the regular one I could just add an assert, sure, but for the other one in cmocka what should I do? Figure out an assert_failure or something similar from within cmocka?

Reply:
If that conundrum is resolved, I will update this PR. And yeah, 74 total instances of `assert` though some came from generated code on my side. Still yeah, many.

Reply:
@kv2019i Do you know how I can actually run the cmocka tests? Any script, manual commands?

Reply:
@paulstelian97 This has the basics https://thesofproject.github.io/latest/developer_guides/unit_tests.html?highlight=cmocka

Reply:
I'd add support for building unit tests to the script. Should I only do it locally or should I make a pull request for *that* too?

Reply:
Dropped the support for unit tests, I will need some help for that one. But that's why this PR is a draft after all.